### PR TITLE
CI build.yml, doc-build.yml: Use output groups

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

With this change, the output of each spkg build is put in foldable output groups.

The output synchronization is not perfect, but it's still much better than before.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: build/bin/sage-logger [hunk 1]
```diff
@@ -89,8 +89,9 @@ else
     # Redirect stdout and stderr to a subprocess running tee.
     # We trap SIGINT such that SIGINT interrupts the main process being
     # run, not the logging.
+
     ( exec 2>&1; eval "$cmd" ) | \
-        ( trap '' SIGINT; tee -a "$logfile" | $SED )
+        ( trap '' SIGINT; if [ -n "$GITHUB_ACTIONS" -a -n "$prefix" ]; then echo "::group::${logname}"; fi; tee -a "$logfile" | $SED; if [ -n "$GITHUB_ACTIONS" -a -n "$prefix" ]; then echo "::endgroup::"; fi )
```
Comment (tobiasdiez) [inline build/bin/sage-logger:94]: Any reason not to use the `ci` env variable?

Comment (mkoeppe) [reply build/bin/sage-logger:94]: What do you mean?

Comment (tobiasdiez) [reply build/bin/sage-logger:94]: Instead of$GITHUB_ACTIONS use $CI

Comment (mkoeppe) [reply build/bin/sage-logger:94]: these output markers are specific to GitHub Actions, no?

Comment (tobiasdiez) [reply build/bin/sage-logger:94]: Also true...

**Review by kwankyu — APPROVED** [review]
OK. Looks nice.

**Comment by mkoeppe** [comment]
Thanks!

**Pull request merged**